arch/arm/src/stm32l5/stm32l5_lse.c

```diff
@@ -0,0 +1,166 @@
+ ****************************************************************************/
+
+#ifdef CONFIG_STM32L5_RTC_LSECLOCK_START_DRV_CAPABILITY
+# if CONFIG_STM32L5_RTC_LSECLOCK_START_DRV_CAPABILITY < 0 || \
+     CONFIG_STM32L5_RTC_LSECLOCK_START_DRV_CAPABILITY > 3
+#  error "Invalid LSE drive capability setting"
+#endif
+#endif
+
+#ifdef CONFIG_STM32L5_RTC_LSECLOCK_RUN_DRV_CAPABILITY
+# if CONFIG_STM32L5_RTC_LSECLOCK_RUN_DRV_CAPABILITY < 0 || \
+     CONFIG_STM32L5_RTC_LSECLOCK_RUN_DRV_CAPABILITY > 3
+#  error "Invalid LSE drive capability setting"
+#endif
+#endif
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: stm32l5_rcc_enablelse
+ *
+ * Description:
+ *   Enable the External Low-Speed (LSE) oscillator and the LSE system clock.
+ *
+ ****************************************************************************/
+
+void stm32l5_rcc_enablelse(void)
+{
+  bool writable;
+  uint32_t regval;
+
+  /* Check if both the External Low-Speed (LSE) oscillator and the LSE system
+   * clock are already running.
+   */
+
+  regval = getreg32(STM32L5_RCC_BDCR);
+
+  if ((regval & (RCC_BDCR_LSEON | RCC_BDCR_LSERDY |
+                 RCC_BDCR_LSESYSEN | RCC_BDCR_LSESYSEN)) !=
+                (RCC_BDCR_LSEON | RCC_BDCR_LSERDY |
+                 RCC_BDCR_LSESYSEN | RCC_BDCR_LSESYSEN))
+    {
+      /* The LSE is in the RTC domain and write access is denied to this
+       * domain after reset, you have to enable write access using DBP bit in
+       * the PWR CR register before to configuring the LSE.
+       */
+
+      writable = stm32l5_pwr_enablebkp(true);
+
+      /* Enable the External Low-Speed (LSE) oscillator by setting the LSEON
+       * bit the RCC BDCR register.
+       */
+
+      regval |= RCC_BDCR_LSEON;
+
+#ifdef CONFIG_STM32L5_RTC_LSECLOCK_START_DRV_CAPABILITY
+      /* Set start-up drive capability for LSE oscillator. */
+
+      regval &= ~RCC_BDCR_LSEDRV_MASK;
+      regval |= CONFIG_STM32L5_RTC_LSECLOCK_START_DRV_CAPABILITY <<
+                RCC_BDCR_LSEDRV_SHIFT;
+#endif
+
+      putreg32(regval, STM32L5_RCC_BDCR);
+
+      /* Wait for the LSE clock to be ready */
+
+      while (((regval = getreg32(STM32L5_RCC_BDCR)) & RCC_BDCR_LSERDY) == 0)
+        {
+          up_waste();
+        }
+
+      /* Enable LSE system clock.  The LSE system clock has been introduced
+       * first by the STM32L5 family of MCUs.  It seems to provide a means
+       * to gate the LSE clock distribution to peripherals.  It must be
+       * enabled for MSI PLL mode (syncing the MSI to the LSE).
+       */
+
+      regval |= RCC_BDCR_LSESYSEN;
+
+      putreg32(regval, STM32L5_RCC_BDCR);
+
+      /* Wait for the LSE system clock to be ready */
+
+      while (!((regval = getreg32(STM32L5_RCC_BDCR)) & RCC_BDCR_LSESYSRDY))
```

Comment:
This can be an infinite loop if the LSECLOCK_RUN_DRV_CAPABILITY is not set high enough.  See https://github.com/apache/incubator-nuttx/pull/2943 and https://github.com/apache/incubator-nuttx/pull/2967 https://github.com/apache/incubator-nuttx/pull/2969

Reply:
Please have a look at https://github.com/apache/incubator-nuttx/pull/2974/commits/8767cf85523812fc7e0cc6de97c1829065a9a98a

Reply:
It is better than the dead wait. But will not ramp to get it started. - I do not feel strongly about ramping herein at this point.